### ISSUE TITLE
[Request] Force sell a coin by some indicator signal, even if DEFAULT_sell_strategy is not true yet (No DCA strategy)

### ISSUE THREAD

**Issue by DmitryBychenko** [taniman/profit-trailer-enhancements#357]
Problem is 
"GAIN is a mandatory sell strategy. This is because without GAIN the other sell strategies can be true while your coin is still not profitable. Having a GAIN strategy allows you to set a minimum profit threshold which must also be true in conjunction with any other strategies."

I'm thinking on creating strategies where you don't need to use DCA. Simple strategy for example:

**Buy:**
DEFAULT_A_buy_strategy = SMACROSS
DEFAULT_A_buy_value = 0.1
DEFAULT_A_buy_value_limit = 0

**Sell:**
DEFAULT_A_sell_strategy = GAIN
DEFAULT_A_sell_value = 2

DEFAULT_B_sell_strategy = SMACROSS
DEFAULT_B_sell_value = -0.1

DEFAULT_trailing_profit = 0.35
DEFAULT_stop_loss_trigger = -1

Currently if there is signal to sell (changing volume, emaspread is too big or too low, smacross, or whatever), bot will not sell, and solution is to watch the bot time after time, and sell coins manually when signal to sell turns true.

Doing this manually contradicts to the idea of automatic trading / bot.

**Expected Behavior**

Bot to sell coins when indicator signal becomes true, no matter what current GAIN value is. For example,

DEFAULT_A_sell_strategy = GAIN
DEFAULT_A_sell_value = 2

DEFAULT_B_sell_strategy = SMACROSS
DEFAULT_B_sell_value = -0.1

DEFAULT_trailing_profit = 0.35
DEFAULT_stop_loss_trigger = -1

Bot should sell the coin even if current **Profit** value =1%, or 0.5%, or -0.5%. If there is no signal - should work normal GAIN sell strategy.

**Current Behavior**
1. If GAIN is > 0, (ex. GAIN = 2).

    a) If **Profit** is > 2% of buy price, PT sells coin after SMACROSS intersection (GAIN = 2 is true, SMACROSS = -0.1 is true)
    b) If **Profit** is < 2% of buy price, PT will not sell the coin after SMACROSS intersection (GAIN = 2 is false, SMACROSS = -0.1 is true)
    c) If **Profit** is < 0% of buy price, PT will not sell the coin after SMACROSS intersection (GAIN = 2 is false, SMACROSS = -0.1 is true)

2. If GAIN is < 0, (ex. GAIN = -1).

DEFAULT_A_sell_strategy = GAIN
DEFAULT_A_sell_value = -1

DEFAULT_B_sell_strategy = SMACROSS
DEFAULT_B_sell_value = -0.1

DEFAULT_trailing_profit = 0.35
DEFAULT_stop_loss_trigger = -1

https://www.dropbox.com/s/8h8qcra4hcq3jmt/gain-1.png?dl=0

In this case coin most likely will be sold by stoploss. One positive deal is done by DEFAULT_trailing_profit = 0.35, when price accidentally dropped out of the range.

https://www.dropbox.com/s/omvsh5d3cggrnhe/gain-1_stoploss.png?dl=0

**Possible Solution**

Create one more command:

DEFAULT_A_sell_strategy = GAIN
DEFAULT_A_sell_value = 2.0

DEFAULT_B_sell_strategy = RSI
DEFAULT_B_sell_value = 65

DEFAULT_trailing_profit = 0.35
DEFAULT_stop_loss_trigger = -1

DEFAULT_A_forcesell_strategy = SMACROSS _(solution)_
DEFAULT_A_forcesell_value = -0.1 _(solution)_

**What we would get:**
1. If there was signal to sell (SMACROSS in this case), bot will sell the coin
2. If there was no signal to sell (SMACROSS in this case), bot would follow normal sell strategy:

DEFAULT_A_sell_strategy = GAIN
DEFAULT_A_sell_value = 2.0

DEFAULT_B_sell_strategy = RSI
DEFAULT_B_sell_value = 65

DEFAULT_trailing_profit = 0.35
DEFAULT_stop_loss_trigger = -1

**In a nutshell**
After coin was purchased, there are 3 possible ways.
1. Coin is sold with profit with normal sell strategy.
2. Coin is sold by indicator signal, which is expected / calculated risk, which, in turn, protects us from more losses.
3. Stop loss.

All these variants can be pre-calculated, which makes overall strategy a bit more predictable. No DCA, but available if needed.

**Comment by angelsomar** [comment]
I just read your topic, publish something similar but thinking about buying and selling, please read it and you want to support it

https://github.com/taniman/profit-trailer-enhancements/issues/361

**Comment by robheath90** [comment]
I have found that I can solve this problem by setting GAIN as a negative value, I trade off MACD day candles which gives me a reliable buy/sell strategy for my long term coins. my GAIN Value is -20% and so my GAIN is nearly always true for all coins. HOWEVER, the big problem with this is that trailing profit ONLY works when a coin is in profit. This means that I am currrently having to manually adjust each coin every time it buys to a fake, low purchase price, in order to allow it to trail.

The simple solution to me would be for trailing_profit to work above the GAIN value specified, in which case people trading on MACD/EMACROSS/SMACROSS could utilise trailing even in a losing condition (in which the coin is indicating a sell position) without having to manually intervene with the bot.

**Comment by aludden** [comment]
I agree, the best would be if it acted like a stoploss. The forcesell would do that.

**Comment by rebolini-pablo** [comment]
> I agree, the best would be if it acted like a stoploss. The forcesell would do that.

+1

**Comment by Masterofcoinak** [comment]
I think this should be closed, as we have now 
https://wiki.profittrailer.com/doku.php?id=pairs.properties#default_sell_strategy_formula
and GAIN is not mandatory anymore.
